spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -303,16 +316,25 @@ protected boolean shouldCopyRequestHeaders() {
 
 		private final MessageHandler delegate;
 
+		private final boolean useNativeEncoding;
+
 		private SendingHandler(MessageHandler delegate, boolean embedHeaders,
-				String[] headersToEmbed) {
+				String[] headersToEmbed, boolean useNativeEncoding) {
 			this.delegate = delegate;
 			this.setBeanFactory(AbstractMessageChannelBinder.this.getBeanFactory());
 			this.embedHeaders = embedHeaders;
 			this.embeddedHeaders = headersToEmbed;
+			this.useNativeEncoding = useNativeEncoding;
 		}
 
 		@Override
 		protected void handleMessageInternal(Message<?> message) throws Exception {
+			Message<?> messageToSend = (this.useNativeEncoding) ? getMessageBuilderFactory().withPayload(message.getPayload()).build() :
```

Comment:
My understanding is that when we use `native` encoding, irrespective of whether we add the `headers` to the outbound messages, the consumer's native de-serializer would ignore those headers and the only headers we would is some `metadata` headers updated at the consumer side (via spring-integration-kafka). Also, I thought this is what we discussed [here] (https://github.com/spring-cloud/spring-cloud-stream/pull/692#discussion_r84076033). What am I missing?